src/include/osl_pvt.h

```diff
@@ -207,6 +207,14 @@ class TypeSpec {
     ///
     int arraylength () const { return m_simple.arraylen; }
 
+    /// Is this a variable length array?
+    ///
+    bool is_varlen_array() const { return m_simple.arraylen < 0; }
+
+    /// Number of elements
+    ///
+    int numelements() const { return std::max (1, arraylength()); }
```

Comment:
Both `arraylength()` and `numelements()` would be invalid to call if `is_varlen_array()` were `true` right? Maybe add a `DASSERT`?